### PR TITLE
Upgrade to mxnet 1.6

### LINKED ISSUE #583: RNNEncoder: "HybridBlock, we do not support mixed NDArrays and Symbols"
I've tried installing mxnet 1.6 (pre-release) and running our tests with it.

There seems to be an issue with the RNNEncoder:

```python
src/gluonts/evaluation/backtest.py:177: in backtest_metrics
    predictor = forecaster.train(train_dataset)
src/gluonts/model/estimator.py:223: in train
    return self.train_model(training_data, validation_data).predictor
src/gluonts/model/estimator.py:208: in train_model
    validation_iter=validation_data_loader,
src/gluonts/trainer/_base.py:297: in __call__
    epoch_loss = loop(epoch_no, train_iter)
src/gluonts/trainer/_base.py:237: in loop
    output = net(*inputs)
../../.virtualenvs/gluon-ts/lib/python3.7/site-packages/mxnet/gluon/block.py:694: in __call__
    out = self.forward(*args)
../../.virtualenvs/gluon-ts/lib/python3.7/site-packages/mxnet/gluon/block.py:1152: in forward
    return self._call_cached_op(x, *args)
../../.virtualenvs/gluon-ts/lib/python3.7/site-packages/mxnet/gluon/block.py:982: in _call_cached_op
    self._build_cache(*args)
../../.virtualenvs/gluon-ts/lib/python3.7/site-packages/mxnet/gluon/block.py:934: in _build_cache
    data, out = self._get_graph(*args)
../../.virtualenvs/gluon-ts/lib/python3.7/site-packages/mxnet/gluon/block.py:926: in _get_graph
    out = self.hybrid_forward(symbol, *grouped_inputs, **params)  # pylint: disable=no-value-for-parameter
src/gluonts/model/seq2seq/_forking_network.py:93: in hybrid_forward
    past_target, feat_static_real, past_feat_dynamic_real
../../.virtualenvs/gluon-ts/lib/python3.7/site-packages/mxnet/gluon/block.py:694: in __call__
    out = self.forward(*args)


self = gluonts.block.encoder.RNNEncoder(bidirectional=True, hidden_size=50, mode="gru", num_layers=1, prefix="encoder_"), x = <Symbol data0>
args = (
[]
<NDArray 0 @cpu(0)>, 
[]
<NDArray 0 @cpu(0)>), has_symbol = True, has_ndarray = True, ctx_set = {cpu(0)}

    def forward(self, x, *args):
        """Defines the forward computation. Arguments can be either
        :py:class:`NDArray` or :py:class:`Symbol`."""
    
        has_symbol, has_ndarray, ctx_set, first_ctx = _gather_type_ctx_info([x] + list(args))
        if has_symbol and has_ndarray:
>           raise ValueError('In HybridBlock, we do not support mixed NDArrays and Symbols'
                             ' types for the input. Please check the type of the args.\n')
E           ValueError: In HybridBlock, we do not support mixed NDArrays and Symbols types for the input. Please check the type of the args.
```

### LINKED ISSUE #671: Tracking: mxnet 1.6
### To update

- [ ] documentation
- [ ] README.md
- [ ] test-dependencies

### Fix

- [ ] https://github.com/awslabs/gluon-ts/issues/583

### Other

- [ ] Update `numpy~1.18`

### PR DESCRIPTION
**Pull request by lostella**
*Issue #, if available:* fixes #583, fixes #671

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,7 +25,7 @@ GluonTS requires Python 3.6, and the easiest
 way to install it is via `pip`:
 
 ```bash
-pip install --upgrade mxnet==1.4.1 gluonts
+pip install --upgrade mxnet>=1.6.0 gluonts
```
Comment (szha) [inline README.md]:
```suggestion
pip install --upgrade "mxnet>=1.6.0,<2" gluonts
```

File: docs/install.rst [hunk 2]
```diff
@@ -9,7 +9,7 @@ command installs the latest version of MXNet.
 
 .. code-block:: console
 
-   pip install --upgrade mxnet>=1.3.1,<1.5.*
+   pip install --upgrade mxnet>=1.6.0
```
Comment (szha) [inline docs/install.rst]:
```suggestion
   pip install --upgrade "mxnet>=1.6.0,<2"
```

**Comment by lostella** [comment]
One issue is that test cases using [this HybridBlock](https://github.com/awslabs/gluon-ts/blob/master/test/support/test_util.py#L129-L132) fail since one of the inputs (`x2[1]`) is not used by the network. I would need the opinion of @canerturkmen on whether this is a problem, or we can remove/modify those test cases.

**Comment by lostella** [comment]
Seems like jitter Cholesky tests need to be disabled on Linux because potrf just segfaults on mxnet 1.6 when the matrix is not positive definite symmetric

**Comment by AaronSpieler** [comment]
This is going to break: https://github.com/awslabs/gluon-ts/pull/678, but overall the images provided in `sagemaker sdk` too, since the MXNet images are `1.4.1` based. The pull request is open and ready to merge for a while now, but since it hasn't been committed it wont show up in the tests.

**Comment by AaronSpieler** [comment]
> This is going to break: #678, but overall the images provided in `sagemaker sdk` too, since the MXNet images are `1.4.1` based. The pull request is open and ready to merge for a while now, but since it hasn't been committed it wont show up in the tests.

Its potentially an easy fix though, once #678 is merged, since we would essentially rely on the pre built SageMaker images, so we could just choose one with an appropriate MXNet version, if there is one.

File: test/support/test_jitter.py [hunk 3]
```diff
@@ -34,6 +35,8 @@
 @pytest.mark.parametrize("jitter_method", ["iter", "eig"])
 @pytest.mark.parametrize("float_type", [np.float32, np.float64])
 def test_jitter_unit(jitter_method, float_type, ctx) -> None:
+    if platform == "linux":
+        pytest.skip(f"skipping this test on {platform}")
```
Comment (jaheba) [inline test/support/test_jitter.py]:
```suggestion
    pytest.skipif(sys.platform == "linux", reason=f"skipping this test on {platform}")
```

File: test/support/test_jitter.py [hunk 4]
```diff
@@ -57,6 +60,8 @@ def test_jitter_unit(jitter_method, float_type, ctx) -> None:
 @pytest.mark.parametrize("jitter_method", ["iter", "eig"])
 @pytest.mark.parametrize("float_type", [np.float32, np.float64])
 def test_jitter_synthetic_gp(jitter_method, float_type, ctx) -> None:
+    if platform == "linux":
+        pytest.skip(f"skipping this test on {platform}")
```
Comment (jaheba) [inline test/support/test_jitter.py]:
```suggestion
    pytest.skipif(sys.platform == "linux", reason=f"skipping this test on {sys.platform}")
```

File: test/support/test_util.py [hunk 5]
```diff
@@ -173,7 +173,10 @@ def hybrid_forward(self, F, x1: Tensor, x2: List[Tensor], my_param):
             y = self.dense_layer(F.broadcast_mul(x2[1], my_param))
             return F.broadcast_add(F.broadcast_mul(x1, x2[0]), y)
 
-    return [TestBlock1, TestBlock2, TestBlock3, TestBlock4]
+    # TODO TestBlock1 is broken when using mxnet 1.6 on linux
```
Comment (jaheba) [inline test/support/test_util.py]: What would be the fix?

File: test/support/test_jitter.py [hunk 6]
```diff
@@ -13,6 +13,7 @@
 
 # Standard library imports
 import math
+from sys import platform
```
Comment (jaheba) [inline test/support/test_jitter.py]:
```suggestion
import sys
```

File: README.md [hunk 7]
```diff
@@ -25,7 +25,7 @@ GluonTS requires Python 3.6, and the easiest
 way to install it is via `pip`:
 
 ```bash
-pip install --upgrade mxnet==1.4.1 gluonts
+pip install --upgrade "mxnet>=1.6.0,<2" gluonts
```
Comment (jaheba) [inline README.md]: Why not just `mxnet==1.6`?

File: test/support/test_jitter.py (hunk 3 above)
Comment (lostella) [reply test/support/test_jitter.py]: nice!

File: README.md (hunk 7 above)
Comment (jaheba) [inline README.md]:
```suggestion
pip install --upgrade mxnet==1.6 gluonts
```

File: docs/install.rst [hunk 8]
```diff
@@ -9,7 +9,7 @@ command installs the latest version of MXNet.
 
 .. code-block:: console
 
-   pip install --upgrade mxnet>=1.3.1,<1.5.*
+   pip install --upgrade "mxnet>=1.6.0,<2"
```
Comment (jaheba) [inline docs/install.rst]:
```suggestion
   pip install --upgrade mxnet==1.6 gluonts
```

**Pull request merged**